worker-sys/src/types/durable_object/state.rs

```diff
@@ -15,4 +15,23 @@ extern "C" {
 
     #[wasm_bindgen(method, js_name=waitUntil)]
     pub fn wait_until(this: &DurableObjectState, promise: &js_sys::Promise);
+
+    #[wasm_bindgen(method, js_name=acceptWebSocket)]
+    pub fn accept_web_socket(this: &DurableObjectState, ws: web_sys::WebSocket);
+
+    #[wasm_bindgen(method, js_name=acceptWebSocket)]
+    pub fn accept_web_socket_with_tags(
+        this: &DurableObjectState,
+        ws: web_sys::WebSocket,
+        tags: Vec<String>,
+    );
+
+    #[wasm_bindgen(method, js_name=getWebSockets)]
+    pub fn get_web_sockets(this: &DurableObjectState) -> Vec<web_sys::WebSocket>;
+
+    #[wasm_bindgen(method, js_name=getWebSockets)]
+    pub fn get_web_sockets_with_tag(
+        this: &DurableObjectState,
+        tag: String,
```

Comment:
Since `&str` gets cloned between Rust and JS anyway I thought it was a more representative type, but upon reflection using `String` causes two allocations, one of the Rust side and one on the JS side. I'll update it.